presto-parquet/src/main/java/com/facebook/presto/parquet/batchreader/decoders/rle/Int64ShortDecimalRLEDictionaryValuesDecoder.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.parquet.batchreader.decoders.rle;
+
+import com.facebook.presto.parquet.batchreader.decoders.ValuesDecoder.ShortDecimalValuesDecoder;
+import com.facebook.presto.parquet.dictionary.LongDictionary;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class Int64ShortDecimalRLEDictionaryValuesDecoder
+        extends Int64RLEDictionaryValuesDecoder
+        implements ShortDecimalValuesDecoder
+{
+    public Int64ShortDecimalRLEDictionaryValuesDecoder(int bitWidth, InputStream in, LongDictionary dictionary)
+    {
+        super(bitWidth, in, dictionary);
+    }
+
+    @Override
+    public void readNext(long[] values, int offset, int length) throws IOException
+    {
+        long[] tempValues = new long[length];
```

Comment:
Allocating a buffer in every readNext call is bad for reliability and performance. There is no need to create a separate decoder and add unnecessary memory allocation and memory copies. I know you're copying existing code, but I suggest just use Int64RLEDictionaryValuesDecoder. If you have time, it would be nice to send a new PR to unify existing RLE dictionary decoders. After all, dictionary IDs can only be RLE/BP encoded, and is not relevant to the data column types. 

Ref: https://parquet.apache.org/docs/file-format/data-pages/encodings/
"Data page format: the bit width used to encode the entry ids stored as 1 byte (max bit width = 32), followed by the values encoded using RLE/Bit packed described above (with the given bit width)."

Reply:
I updated this to just use the Int64 decoder